### DISCUSSION TITLE
Unable to use iperf3 on UDP with external app

### DISCUSSION THREAD

**Discussion by volm92** [InterDigitalInc/AdvantEDGE#298] [Q&A]
Hi!
I can't get iperf3 in UDP mode to work with advantedge: iperf3 works fine with TCP, but with UDPit can't connect to the server.
This is the scenario:
![image](https://user-images.githubusercontent.com/80367413/131871531-9b10aa2f-62ea-4f2c-a7de-7a6a9554ed12.png)
And these are the settings:
![image](https://user-images.githubusercontent.com/80367413/131871875-3c0a6558-da04-4e82-b1f1-09ec7b169c24.png)
and 
![image](https://user-images.githubusercontent.com/80367413/131871957-0c4653d3-30ff-4a15-917a-59caf3157625.png)

When I try to use iperf3 I write: 
`iperf3 -s -p 30141` for the server (it is receiving so the output port)
and:
`iperf3 -c 192.168.178.145 -p 30171 -u` for the client (it is sending so the input port)

What is wrong?
Using the same settings but with TCP it is working!

Thanks!

**Comment by roymx** [comment]
Hello @volm92 ,

I am surprised by your configuration, typically:
- _ingress rule_ is specified for _terminal application_
- _egress  rule_ is specified for the _edge application_

And the flow is
![image](https://user-images.githubusercontent.com/48520190/141149799-3fec2197-6ac1-4e5a-a350-7f497539a3fe.png)

As for Iperf3 not connecting, I'm not sure.
There may be some control information exchanged between the client and server before the traffic flow is established.
You should verify this with a Wireshark capture.
